### DISCUSSION TITLE
Sonar Coverage

### DISCUSSION THREAD

**Discussion by tnf-shivanggupta** [sveltejs/kit#10854] [Q&A]
I have to get the sonar coverage for my sveltekit project but sonar cloud is not considering .svelte files

**Comment by thomas-gg** [comment]
Hey @tnf-shivanggupta! Since Sonar cloud is not providing a solution for .svelte files (and possibly Vue etc), a custom work-around would be to create a post-processing solution or a custom coverage reporter (for Istanbul) for Jest/Vitest. A custom coverage reporter discussion occurred within this Github Issue: https://github.com/jestjs/jest/issues/9112#issuecomment-1284058331. I created a quick example of what that may look like [here](https://github.com/thomas-gg/generic-sonar-code-coverage).

You can also create a post-processing solution based on [lcov](https://github.com/jestjs/jest/issues/9112) using javascript or [cobertura reports](https://github.com/softreigns/coverage-report-cobertura-to-sonar-generic/blob/main/cobertura-sonar-generic.xslt) using XSLT. These solutions are not limited to using Istanbul as a code coverage reporter, but may get more complex depending on your repo.

Please let me know if you've thought of a better solution! Thanks
